### PR TITLE
Jest: Setup jest to run with Playwright 

### LINKED ISSUE #13: Setup Jest to run with Playwright
## TODO

- [ ] Figure out a way to run Jest with Playwright

- [ ] Test the sample script in Playwright with Jest

### PR DESCRIPTION
**Pull request by hariclerry**
### Tickets
Closes #13, #5 

### What does this PR do?
- This PR setup jest to run with Playwright

### What are the changes made?
- Installed jest
- Installed jest-playwright-preset
- Added configuration for jest and playwright

### Screen shots
None

### CODE REVIEW

File: tests/playwright/jest-playwright.config.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+	launchOptions: {
+		headless: false
```
Comment (sovoid) [inline tests/playwright/jest-playwright.config.js]: This should not be hardcoded. We should essentially set an environment variable which if not exists runs in headless by default

We should add an environment variable like HEADLESS which if set to 'true' should make the test headless

headless: (HEADLESS && HEADLESS === 'true')

File: tests/playwright/jest.e2e.config.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+	preset: 'jest-playwright-preset',
+	testMatch: [ '**/tests/playwright/specs/**/*.js' ],
```
Comment (sovoid) [inline tests/playwright/jest.e2e.config.js:3]: nice job done fixing this

File: tests/playwright/specs/wikipediaHomePage.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+		browser = await chromium.launch();
+		page = await browser.newPage();
+		response = await page.goto( 'https://www.wikipedia.org/' );
+	} );
+
+	afterAll( async () => {
+		await browser.close();
+	} );
+
+	it( 'test if page loads successfully', async () => {
```
Comment (sovoid) [inline tests/playwright/specs/wikipediaHomePage.js]: The describe and the test together should form a language...for example

the `describe` block says "Wikipedia Home Page"

so the `it` block should read "should load successfully"

File: tests/playwright/specs/wikipediaHomePage.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+		expect( response.status() ).toBe( 200 );
+	} );
+
+	it( 'should check if page sub title is correct', async () => {
```
Comment (sovoid) [inline tests/playwright/specs/wikipediaHomePage.js]: Same as above

**Review by sovoid — CHANGES_REQUESTED** [review]

File: package.json [hunk 5]
```diff
@@ -11,7 +11,8 @@
     "postdoc": "grunt copy:jsduck",
     "selenium": "bash ./tests/selenium/selenium.sh",
     "selenium-daily": "MW_SCRIPT_PATH=/w npm run selenium-test -- --mochaOpts.grep @daily",
-    "selenium-test": "wdio ./tests/selenium/wdio.conf.js"
+    "selenium-test": "wdio ./tests/selenium/wdio.conf.js",
+    "jest-test": "jest -c ./tests/playwright/jest.e2e.config.js"
```
Comment (sovoid) [inline package.json]: Add scripts:

`playwright:open`: Run in non-headless mode
`playwright:close`: Run in headless mode

File: tests/playwright/specs/wikipediaHomePage.js (hunk 4 above)
Comment (sovoid) [reply tests/playwright/specs/wikipediaHomePage.js]: Make a single test which checks only the subtitle as if the subtitle loads it implies that the page has loaded correctly

**Review by sovoid — CHANGES_REQUESTED** [review]

File: tests/playwright/jest.e2e.config.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+	preset: 'jest-playwright-preset',
```
Comment (sovoid) [inline tests/playwright/jest.e2e.config.js:2]: check where this should be added

File: tests/playwright/jest-playwright.config.js (hunk 1 above)
Comment (sovoid) [inline tests/playwright/jest-playwright.config.js]: This is not being read from the configuration file oddly. Please verify  if the config is correct

**Comment by hariclerry** [comment]
@und3fined-v01d, I have implemented the requested changes. Please review and let me know if everything is fine. Thanks

File: tests/playwright/README.md [hunk 7]
```diff
@@ -0,0 +1,9 @@
+# playwright tests
+
+## Getting started
+
+- Run `npm ci`
```
Comment (sovoid) [inline tests/playwright/README.md]: Update README and add optional environment variables that can be set along with their defaults

File: tests/playwright/README.md [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+- Run `node tests/playwright/specs/wikipediaHomePage.js` from the app root directory
+
+### Run test with Jest
+- Run `npm run jest-test`
```
Comment (sovoid) [inline tests/playwright/README.md]: update this command to new one and add a note that it runs headless by default

**Review by sovoid — CHANGES_REQUESTED** [review]

**Review by vidhi-mody — APPROVED** [review]

**Review by sovoid — APPROVED** [review]

**Pull request merged**